### PR TITLE
feature: add modal drag config

### PR DESCRIPTION
**Pull request by youpengly**
* feature: modal drag config-- nzDraggable
* fix bug [#120](https://github.com/NG-ZORRO/ng-zorro-antd/issues/120 )
* delete 5 commits, master rebase 
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[x] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
* click modal header to drag

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by vthinkxie** [comment]
Hi @youpengly , please add test.

**Comment by youpengly** [comment]
@vthinkxie , ok , I will add test later. 
May I ask you a question:  what I have to do is just  add test code in nz-modal.spec.ts for the new updated code ?

**Comment by vthinkxie** [comment]
@youpengly after that ,we will review your code.

**Comment by youpengly** [comment]
@vthinkxie , ok , I tested it manually  in chrome ,firefox and edge, right now I  have poor experience in writting test case. anyway, I will try my best to handle it.

**Comment by vthinkxie** [comment]
@youpengly you have to rebase on master when have conflict with it.

**Comment by youpengly** [comment]
@vthinkxie ok, I have delete the previous 5 commits and rebase master. thanks for your patience to answer

**Comment by youpengly** [comment]
@vthinkxie,  
*  Right now first allow me to resolve the conflict file  nz-modal.component.ts (this time it auto merge?) ?
      and later rebase my master branch?
*  How dose I know you have reviewed my code?
Thank you very much.

**Comment by vthinkxie** [comment]
@youpengly 
1.rebase之后再 push -f ，pull request会自动更新，不需要你手动在页面上 resolve conflicts
2.我们会在所有ci完全通过后参与 code review

**Comment by ppjjzz** [comment]
any update for this pr?

**Comment by youpengly** [comment]
@ppjjzz, The last commit,
 I have already rebase and push -f, but this pr dose not seem to auto update(or it alreay updated but dose not pass the ci) ,I don't know the reason.  
I'm really sorry about that.

**Comment by vthinkxie** [comment]
@youpengly the test is not strong enough, you have to add more test case for your code to increase the coverage to more than 95.98%, which is only 53% now.

**Comment by youpengly** [comment]
@vthinkxie  Thanks for your detailed explanation, previous I don't know why the reason.

File: components/modal/nz-modal.component.html [hunk 1]
```diff
@@ -40,7 +40,7 @@
 
 <!-- [Predefined] Default Modal Content -->
 <ng-template #tplContentDefault>
-  <div *ngIf="nzTitle" class="ant-modal-header">
+  <div #modalHeader *ngIf="nzTitle" class="ant-modal-header" (mousedown)="initDrag($event)" [ngStyle]="{'cursor': cursorStyle}">
```
Comment (wilsoncook) [inline components/modal/nz-modal.component.html]: Be care the `modalHeader` may be not exist when `nzTitle` is empty, this can cause side effect errors

**Comment by vthinkxie** [comment]
Due to no progress for a long time, this issue will be closed. Please contact us if you have any questions.

**Pull request closed without merging**